### ISSUE TITLE
ENH: Add preload=str to concatenate_raw and concatenate_epochs

### ISSUE THREAD

**Issue by melyndaschreiber** [mne-tools/mne-python#6380]
Hello all, I am using mne.concatenate_raws to collect all my raw data for a single subject. This often caps out my memory on my computer. Is there a way to better manage memory in this situation. Another potential solution is to concatenate epochs which is much better for memory. However, I am concerned that filtering short epochs would cause other issues in my data.

@larsoner has suggested the following:
can you open an issue about concatenate_raws supporting preload as a string?
basically it would allow using a memmap'ed file
in the meantime you can filter each raw instance and save it to disk
and load each of those filtered ones with preload=False (don't load into memory)
then do concatenate_raws on those
then do mne.Epochs on the concatenated instance
the epochs will pull data from the concatenated instance, which will pull them from disk instead of memory
